sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -839,8 +839,6 @@ case class RepartitionByExpression(
 
   require(numPartitions > 0, s"Number of partitions ($numPartitions) must be positive.")
 
-  require(partitionExpressions.nonEmpty, "At least one partition-by expression must be specified.")
```

Comment:
Just for safety, also keep this change?

Reply:
That would change the current behavior of '.repartition(numPartitions, Seq.empty: _*)' and I'd like to avoid that.

In fact, I've just raised a separate ticket about the latter.